src/AccelStepperFirmata.cpp

```diff
@@ -0,0 +1,430 @@
+        } else if (wireCount == 4 && stepType == STEP_TYPE_WHOLE) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::FULL4WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3, motorPin4, false);
+        } else if (wireCount == 4 && stepType == STEP_TYPE_HALF) {
+          stepper[deviceNum] = new AccelStepper(AccelStepper::HALF4WIRE, stepOrMotorPin1, directionOrMotorPin2, motorPin3, motorPin4, false);
+        }
+
+        
+        // If there is still another byte to read we must be inverting some pins
+        if (argc >= index) {
+          invertPins = argv[index];
+          if (wireCount == 1) {
+             stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 4 & 0x01);
+          } else {
+            stepper[deviceNum]->setPinsInverted(invertPins & 0x01, invertPins >> 1 & 0x01, invertPins >> 2 & 0x01, invertPins >> 3 & 0x01, invertPins >> 4 & 0x01);
+          }
+        }
+
+        if (interface & 0x01) {
+          stepper[deviceNum]->setEnablePin(enablePin);
+        }
+
+        /*
+          Default to no acceleration. We set the acceleration value high enough that our speed is 
+          reached on the first step of a movement. 
+          More info about this hack in STEPPER_SET_ACCELERATION.
+
+          The lines where we are setting the speed twice are necessary because if the max speed doesn't change 
+          from the default value then our time to next step does not get computed after raising the acceleration.
+        */
+        stepper[deviceNum]->setMaxSpeed(2.0);
+        stepper[deviceNum]->setMaxSpeed(1.0);
+        stepper[deviceNum]->setAcceleration(MAX_ACCELERATION);
+
+        isRunning[deviceNum] = false;
+
+      }
+
+      else if (stepCommand == STEPPER_STEP) {
+
+        numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->move(numSteps);
+          isRunning[deviceNum] = true;
+        }
+
+      }
+
+      else if (stepCommand == STEPPER_ZERO) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setCurrentPosition(0);
+        }
+      }
+
+      else if (stepCommand == STEPPER_TO) {
+        if (stepper[deviceNum]) {
+          numSteps = decode32BitSignedInteger(argv[2], argv[3], argv[4], argv[5], argv[6]);
+          stepper[deviceNum]->moveTo(numSteps);
+          isRunning[deviceNum] = true;
+        }
+      }
+
+      else if (stepCommand == STEPPER_ENABLE) {
+        if (stepper[deviceNum]) {
+          if (argv[2] == 0x00) {
+            stepper[deviceNum]->disableOutputs();
+          } else {
+            stepper[deviceNum]->enableOutputs();
+          }
+        }
+      }
+
+      else if (stepCommand == STEPPER_STOP) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->stop();
+          isRunning[deviceNum] = false;
+          reportPosition(deviceNum, true);
```

Comment:
Should the 2nd param here be `true` or `false`. Currently this will send a `MOVE_COMPLETE` message. However is it also assumed a user can resume a stopped stepper? If so then the 2nd param should be `false` and this would then send a `REPORT_POSITION` message.

Reply:
Ok, I read up on `AccelStepper.stop()`. It is actually terminal (the user cannot resume a step sequence after calling stop). Therefore, we can simply remove `reportPosition(deviceNum, true);` here since the `STEPPER_COMPLETE` message will be sent from the `update()` method after a `STEPPER_STOP` message is received. In that case `SETPPER_COMPLETE` means the sequence has either completed or was stopped (terminated at the current step count).

Reply:
Hummm, I noticed something else interesting here in experimenting further. Do we want stop() to stop immediately or do we want stop() to ramp down. The behavior in AccelStepper according to the documentation is actually to ramp down as quickly as possible using the current speed and acceleration parameters. If you comment out `isRunning[deviceNum] = false` and `reportPosition(deviceNum, true)`, you'll see that behavior if you run something like this in JS:

```javascript
board.accelStepperSpeed(0, 400);
board.accelStepperAcceleration(0, 100);

board.accelStepperTo(0, 2000, function(position) {
  console.log("Current position: " + position);
});

setTimeout(function() {
  // we need to decide if we want this to stop immediately or if we want it to ramp
  // down if acceleration was enabled
  board.accelStepperStop(0);
}, 2000);
```

Reply:
Here's an example from the AccelStepper documentation: http://www.airspayce.com/mikem/arduino/AccelStepper/Quickstop_8pde-example.html. There they allow the motor to decelerate quickly after the stop() function.

Reply:
I don't want to presume I know better than Mike, but I would think stop() should be immediate (for example, when a limit switch is hit). I know enable(false) would give us that immediate behavior, but not everyone will have their enable pins wired up.